lib/hyacinth/adapters/digital_object_search_adapter/solr/document_generator.rb

```diff
@@ -48,11 +48,18 @@ def merge_title_fields_for(digital_object, solr_document = {})
         end
 
         def merge_title_fields_for!(digital_object, solr_document = {})
-          solr_document['sort_title_ssi'] = digital_object.generate_title(true)
-          indexable_title = digital_object.generate_title
-          solr_document['title_ss'] = indexable_title
-          add_keywords(indexable_title, solr_document)
-          add_titles(indexable_title, solr_document)
+          display_label = digital_object.generate_label
+          solr_document['displayLabel_ss'] = display_label
+          if digital_object.title.present?
+            solr_document['title_sortPortion_ssi'] = digital_object.title.dig('value', 'sort_portion')
+            solr_document['sort_title_ssi'] = digital_object.title.dig('value', 'sort_portion')
+            solr_document['title_nonSortPortion_ssi'] = digital_object.title.dig('value', 'non_sort_portion')
```

Comment:
```suggestion
            solr_document['title_sort_portion_ssi'] = digital_object.title.dig('value', 'sort_portion')
            solr_document['sort_title_ssi'] = digital_object.title.dig('value', 'sort_portion')
            solr_document['title_non_sort_portion_ssi'] = digital_object.title.dig('value', 'non_sort_portion')
```
Suggesting snake_case for key consistency.

Reply:
It's not tested (!), but Hyacinth::DigitalObject::SolrKeys actually camelcases the string keys (as does the Rights indexing) - this ensures no unexpected collisions here. If we want to revisit that,, we should ticket separately. I'll add a test for the SolrKeys behavior, though!

Reply:
Oh, you're right!  Comment withdrawn for now then!  Maybe worth revisiting later.

Reply:
And thank you for adding the test!